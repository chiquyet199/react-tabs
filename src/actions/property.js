import API from "../utils/api";
import {errorNotification} from "./notifications";
import {
  BRANCH_SELECTED,
  BRANCH_PROPERTIES_RECEIVED,
  CLEAR_BRANCH_PROPERTIES,
} from "./types";
import {extractBranch} from "../utils/properties";

// {params} contains branchId along with sort params
export const getPropertyList = params => async (dispatch, getState) => {
  let properties = [];
  try {
    properties = await API.getProperties(params);
    dispatch({
      type: BRANCH_SELECTED,
      data: extractBranch(getState().agencyBranches.items, properties),
    });
    dispatch({type: BRANCH_PROPERTIES_RECEIVED, data: properties});
  } catch (err) {
    dispatch(errorNotification(err));
  }

  return properties;
};

export const clearProperties = () => async dispatch => {
  dispatch({type: CLEAR_BRANCH_PROPERTIES});
};
