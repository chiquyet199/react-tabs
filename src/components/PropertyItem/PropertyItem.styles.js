import {themed} from "../../utils/common";
import styledNative from "styled-components/native";
import DefaultPropertyThumbnail from "../svg/DefaultPropertyImage";

export const PropertyItemWrapper = themed(styledNative.View`
  align-items: center;
  border-bottom-color: ${({theme, globaltheme}) =>
    theme[globaltheme].boxBorder};
  border-bottom-width: ${({noBorder}) => (noBorder ? 0 : "1px")};
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  margin: 0 17px;
  padding: 17px 0;
`);

export const PropertyImage = styledNative.Image`
  height: 46px;
  margin-right: 10px;
  width: 46px;
  border-radius: 3px;
`;

export const DefaultPropertyImageWrapper = styledNative.View`
  align-items: center;
  display: flex;
  height: 46px;
  justify-content: center;
  margin-right: 10px;
  width: 46px;
`;

export const StyledDefaultPropertyThumbnail = styledNative(
  DefaultPropertyThumbnail,
)`
  width: 46px;
  height: 46px;
`;

export const PropertyAddress = styledNative.View`
  display: flex;
  flex-direction: column;
  margin-right: 10px;
  flex-grow: 2;
`;
export const PropertyAddressLine = themed(styledNative.Text`
  color: ${({theme, globaltheme}) => theme[globaltheme].textColor};
  font-size: 16px;
  width: 100%;
`);
export const PropertyRent = themed(styledNative.Text`
  color: ${({theme, globaltheme}) => theme[globaltheme].textColor};
  font-size: 16px;
  font-weight: bold;
  text-align: right;
`);

export const ToggleIcon = styledNative.Image`
  width: 32px;
  height: 32px;
`;
