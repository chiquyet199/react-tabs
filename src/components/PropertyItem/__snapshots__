/Users/edward/Desktop/canopy-frontend/packages/canopy-frontend-common/src/components/PropertyItem/__native_snapshots__