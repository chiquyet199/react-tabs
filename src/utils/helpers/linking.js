import {Linking} from "react-native";

/*
this wrapping nonsense is necessary because jest won't mock
the Linking library in react-native directly on runs of
web tests
*/
export const openURL = url => Linking.openURL(url);
