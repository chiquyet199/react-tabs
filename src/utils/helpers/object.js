export const safe = obj => {
  return new Proxy(obj, {
    get: (target, name) => {
      if (Object.keys(target).includes(name)) {
        const result = target[name];

        return result instanceof Object ? safe(result) : result;
      }

      return safe({});
    },
  });
};
