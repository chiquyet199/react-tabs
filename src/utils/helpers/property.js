import moment from "moment";

export const getPrimaryPropertyImage = propertyImages => {
  if (propertyImages && propertyImages.length) {
    const primaryImage = propertyImages.find(img => img.isPrimary);

    return (primaryImage && primaryImage.url) || propertyImages[0].url;
  }

  return null;
};

const getRentIncrementUnit = rentFrequency => {
  switch (rentFrequency) {
    case "YEARLY":
      return "y";
    case "MONTHLY":
      return "M";
    default:
      return "w";
  }
};

const getRentIncrementValue = rentFrequency => {
  switch (rentFrequency) {
    case "FOUR_WEEKLY":
      return 4;
    case "FORTNIGHTLY":
      return 2;
    default:
      return 1;
  }
};

const isWorkingDay = dayToTest => dayToTest.day() > 0 && dayToTest.day() < 6;

const getNextWorkingDay = dayToMove => {
  if (!isWorkingDay(dayToMove)) {
    if (dayToMove.day() === 0) {
      return moment(dayToMove).add(1, "d");
    }
    if (dayToMove.day() === 6) {
      return moment(dayToMove).add(2, "d");
    }
  }

  return moment(dayToMove);
};

export const getRentDueDate = (lease, today = Date.now()) => {
  const {
    duration: {startDate},
    rentFrequency,
  } = lease;
  const todayAsMoment = moment(today);
  const increment = getRentIncrementValue(rentFrequency);
  const incrementUnit = getRentIncrementUnit(rentFrequency);
  let x = 1;

  while (
    moment(startDate)
      .add(x * increment, incrementUnit)
      .isSameOrBefore(todayAsMoment)
  ) {
    x += 1;
  }

  const rentDueDate = moment(startDate).add(x * increment, incrementUnit);

  if (!isWorkingDay(rentDueDate)) {
    return getNextWorkingDay(rentDueDate).format("YYYY-MM-DD");
  }

  return rentDueDate.format("YYYY-MM-DD");
};
