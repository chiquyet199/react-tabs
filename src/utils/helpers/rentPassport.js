import React, {Component, Fragment} from "react";
import styled from "styled-components/native";
import {extend, flatten} from "underscore";
import {colors} from "../../constants/theme";
import {RENT_PASSPORT_STATUSES} from "../../constants/statuses";
import Icon from "../../components/Icon";

export const TextWrapper = styled.View`
  flex-grow: 3;
  padding-left: 15px;
  flex-direction: column;
  align-items: flex-start;
  padding: 10px;
`;

export const StyledText = styled.Text`
  color: ${({color}) => color || colors.canopySteel};
  flex-basis: 100%;
  font-weight: normal;
  margin-bottom: 0;
`;

export const StyledLink = styled.Text`
  color: ${({color}) => color || colors.canopyGreen};
  flex-basis: 100%;
  font-weight: bold;
  margin-bottom: 0;
`;

export const generateNotificationType = (type, iconName, color) => {
  return {
    type,
    iconName,
    textColor: color,
    iconColor: color,
  };
};

export const generateNotificationProps = spec => {
  return extend(
    {
      textColor: null,
      linkColor: null,
      action: () => {},
      iconNameOverride: null,
      keys: [],
    },
    spec,
  );
};

export const calculateAffordability = rentPassport => {
  let affordabilityFromAnnualIncome = 0;
  let affordabilityFromDisposableIncomeAndRent = 0;
  if (rentPassport && rentPassport.financial) {
    const {financial} = rentPassport;
    if (financial.grossAnnualIncome) {
      const {grossAnnualIncome} = financial;

      affordabilityFromAnnualIncome = Math.round(
        (grossAnnualIncome * 0.4) / 12,
      );
    }

    if (financial.disposableIncome && financial.monthlyRentPayment) {
      const {disposableIncome, monthlyRentPayment} = financial;

      affordabilityFromDisposableIncomeAndRent = Math.round(
        disposableIncome + monthlyRentPayment,
      );
    }
  }

  return Math.max(
    affordabilityFromAnnualIncome,
    affordabilityFromDisposableIncomeAndRent,
  );
};

export default class RentPassportHelpers extends Component {
  static getStatusFromMultipleStatuses(component) {
    const flat = comp =>
      [].concat(...Object.values(comp).map(el => el.map(obj => obj.status)));
    const statuses = flatten(flat(component.check || component));
    const validStatuses = ["RED", "YELLOW", "GREEN"];
    for (let i = 0; i < validStatuses.length; i += 1) {
      if (statuses.includes(validStatuses[i])) {
        return validStatuses[i];
      }
    }

    return RENT_PASSPORT_STATUSES.none;
  }

  static getMultipleStatuses(component) {
    const flat = comp =>
      [].concat(
        ...Object.values(comp).map(el =>
          el.map(obj => [{status: obj.status, errors: obj.messages}]),
        ),
      );

    return flatten(flat(component.check || component.component || component));
  }

  static NotificationTypes = {
    GREEN: generateNotificationType(
      "rent_passport_info",
      "tick-circle",
      colors.canopyGreen,
    ),
    YELLOW: generateNotificationType(
      "rent_passport_warning",
      "warning",
      colors.orange,
    ),
    RED: generateNotificationType(
      "rent_passport_error",
      "close-circle",
      colors.alertRed,
    ),
  };
  static NotificationProps = actions => [
    generateNotificationProps({
      textColor: colors.orange,
      linkColor: colors.orange,
      keys: [
        "pendingReference",
        "pendingEmployerReference",
        "pendingSourcesOfIncome",
        "previousInsolvencies_warning",
        "previousCCJs_warning",
      ],
    }),
    generateNotificationProps({
      textColor: colors.orange,
      linkColor: colors.canopyGreen,
      action: () => {
        actions.goTo("/help");
      },
      keys: [
        "unverified",
        "dateOfBirth_unverified",
        "nationality_unverified",
        "name_unverified",
        "location_currentAddress_pendingManualVerification",
      ],
    }),
    generateNotificationProps({
      textColor: colors.alertRed,
      linkColor: colors.canopyGreen,
      action: () => actions.goTo("/help"),
      keys: [
        "watchList_politicallyExposedPerson_failed",
        "identity_mortality_failed",
        "watchList_legalAndRegulatoryWarnings_failed",
        "watchList_sanction_failed",
      ],
    }),
    generateNotificationProps({
      textColor: colors.orange,
      linkColor: colors.canopyGreen,
      action: () => actions.goTo("/request-reference"),
      keys: ["requestReference"],
    }),
    generateNotificationProps({
      textColor: colors.orange,
      linkColor: colors.canopyGreen,
      action: () => actions.goTo("/upload-documents/verify-income"),
      keys: [
        "finance_grossAnnualIncome_pendingManualVerification",
        "sourcesOfIncome_unverified",
        "grossAnnualIncome_unverified",
      ],
    }),
    generateNotificationProps({
      textColor: colors.orange,
      linkColor: colors.canopyGreen,
      action: () => actions.goTo("/renting-with-guarantor"),
      iconNameOverride: "close-circle",
      keys: ["paidWork_unverified"],
    }),
    generateNotificationProps({
      textColor: colors.alertRed,
      linkColor: colors.canopyGreen,
      action: () => actions.goTo("/renting-with-guarantor"),
      iconNameOverride: "close-circle",
      keys: [
        "credit_insolvency_status_failed",
        "credit_countyCourtJudgments_failed",
        "credit_insolvency_failed",
        "credit_numberOfBankruptcies_failed",
        "creditScore_failed",
      ],
    }),
    generateNotificationProps({
      textColor: colors.orange,
      linkColor: colors.canopyGreen,
      iconNameOverride: "warning",
      keys: [
        "Tenant_has_less_eligible_",
        "Tenant_is_smoker_",
        "Tenant_has_pets_",
      ],
    }),
  ];

  static defaultNotificationProp = generateNotificationProps(
    colors.canopySteel,
    colors.canopySteel,
  );

  static getNotificationType(status) {
    const validTypes = Object.keys(this.NotificationTypes);

    if (validTypes.includes(status)) {
      return this.NotificationTypes[status];
    }

    return generateNotificationType(
      "rent_passport_error",
      "help",
      colors.canopySteel,
    );
  }

  static getNotificationProps(messageKey, actions) {
    const propsList = this.NotificationProps(actions);

    return (
      propsList.find(prop => prop.keys.includes(messageKey)) ||
      this.defaultNotificationProp
    );
  }

  static renderContent(status, message, errorsText, actions) {
    if (errorsText) {
      const messageKey = message.replace(/[.!\s]/gi, "_");
      const {
        iconNameOverride,
        textColor,
        linkColor,
        action,
      } = this.getNotificationProps(messageKey, actions);

      const {iconName, iconColor} = this.getNotificationType(status);
      const texts =
        errorsText && errorsText[messageKey]
          ? errorsText[messageKey]
          : messageKey;

      return (
        <Fragment>
          <Icon
            name={iconNameOverride || iconName}
            color={textColor || iconColor}
            size={26}
          />
          <TextWrapper>
            <StyledText color={textColor || iconColor}>{texts.text}</StyledText>
            {texts.action && (
              <StyledLink color={linkColor} onPress={action}>
                {texts.action}
              </StyledLink>
            )}
          </TextWrapper>
        </Fragment>
      );
    }

    return null;
  }
}
