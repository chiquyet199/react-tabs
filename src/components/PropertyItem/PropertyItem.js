import React, {Component} from "react";
import {connect} from "react-redux";
import PropertySpec from "../../utils/api/specs/PropertySpec";
import {Collapse, CollapseContent} from "../Collapse";
import {func, shape, string, element, bool} from "prop-types";
import {View, TouchableOpacity} from "react-native";
import upArrow from "../../assets/ic_up_arrow_gray/ic_up_arrow_gray.png";
import downArrow from "../../assets/ic_down_arrow_gray/ic_down_arrow_gray.png";
import {
  ToggleIcon,
  PropertyImage,
  PropertyItemWrapper,
  PropertyAddress,
  PropertyAddressLine,
  StyledDefaultPropertyThumbnail,
  PropertyRent,
  DefaultPropertyImageWrapper,
} from "./PropertyItem.styles.js";
import {getPrimaryPropertyImage} from "../../utils/helpers/property";

export class PropertyItem extends Component {
  constructor() {
    super();

    this.state = {
      isCollapsed: false,
    };
    this.onToggle = this.onToggle.bind(this);
  }

  onToggle() {
    const {isCollapsed} = this.state;

    this.setState({isCollapsed: !isCollapsed});
  }

  getPropertyImage = propertyImages => {
    const uri = getPrimaryPropertyImage(propertyImages);
    if (uri) {
      return <PropertyImage source={{uri}} />;
    }

    return (
      <DefaultPropertyImageWrapper>
        <StyledDefaultPropertyThumbnail />
      </DefaultPropertyImageWrapper>
    );
  };

  handlePress = () => {
    const {data, itemClick} = this.props;
    itemClick(data.id);
  };

  render() {
    const {
      itemClick,
      data,
      translations,
      collapseContent,
      compact,
    } = this.props;
    const {isCollapsed} = this.state;
    const output = (
      <PropertyItemWrapper compact={compact}>
        {this.getPropertyImage(data.propertyImages || [])}
        <PropertyAddress>
          <View>
            <PropertyAddressLine>{data.address.line1}</PropertyAddressLine>
          </View>
          <View>
            <PropertyAddressLine>
              {data.address.city || data.address.town}
            </PropertyAddressLine>
          </View>
          <View>
            <PropertyAddressLine>{data.address.postCode}</PropertyAddressLine>
          </View>
        </PropertyAddress>
        <PropertyRent>
          {collapseContent ? (
            <ToggleIcon source={isCollapsed ? downArrow : upArrow} />
          ) : (
            `${translations.currency_symbol}${data.rentAmount}${
              translations.rental_period_abbreviations[
                data.rentFrequency.toLowerCase()
              ]
            }`
          )}
        </PropertyRent>
      </PropertyItemWrapper>
    );

    if (collapseContent) {
      return (
        <Collapse
          collapsed={isCollapsed.toString()}
          onToggle={this.onToggle}
          collapsedByDefault
        >
          <CollapseContent type="header">{output}</CollapseContent>
          <CollapseContent type="body">{collapseContent}</CollapseContent>
        </Collapse>
      );
    }

    if (itemClick) {
      return (
        <TouchableOpacity onPress={this.handlePress}>{output}</TouchableOpacity>
      );
    }

    return output;
  }
}

PropertyItem.propTypes = {
  data: PropertySpec,
  translations: shape({
    rental_period_abbreviations: shape({
      weekly: string.isRequired,
      fortnightly: string.isRequired,
      four_weekly: string.isRequired,
      monthly: string.isRequired,
      yearly: string.isRequired,
    }),
    currency_symbol: string.isRequired,
  }),
  itemClick: func,
  collapseContent: element,
  compact: bool,
};

const mapStateToProps = state => ({
  translations: {
    rental_period_abbreviations:
      state.locale.translations.property_list.rental_period_abbreviations,
    currency_symbol: state.locale.translations.common.currency_symbol,
  },
});

export default connect(mapStateToProps)(PropertyItem);
