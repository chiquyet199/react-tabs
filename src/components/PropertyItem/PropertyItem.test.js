import React from "react";
import {Platform, TouchableOpacity} from "react-native";
import {
  PropertyImage,
  PropertyItemWrapper,
  PropertyAddressLine,
  StyledDefaultPropertyThumbnail,
  PropertyRent,
} from "./PropertyItem.styles.js";
import {shallow} from "enzyme";
import {extend} from "underscore";
import {
  getMockProperties,
  getMockTranslations,
  getMockPropertiesAsArray,
} from "../../__mocks__/properties";
import {PropertyItem} from "./PropertyItem";
import {getPrimaryPropertyImage} from "../../utils/helpers/property";

jest.mock("../../utils/helpers/property");

describe("Property item tests", () => {
  beforeEach(() => {
    getPrimaryPropertyImage.mockClear();
    getPrimaryPropertyImage.mockImplementation(
      () => "http://www.findyourcanopy.com/img124.jpeg",
    );
  });

  it("PropertyItem snapshot test on - " + Platform.OS, () => {
    expect(getComponent()).toMatchSnapshot();
  });

  describe("Property Image", () => {
    it("displays image source passed from getPrimaryPropertyImage", () => {
      const properties = getMockProperties();
      const testCases = [
        [
          properties.hasPrimaryImageNotAtZero,
          "http://www.findyourcanopy.com/image101.jpeg",
        ],
        [
          properties.hasPrimaryImageAtZero,
          "http://www.findyourcanopy.com/image202.jpeg",
        ],
        [
          properties.hasMixOfImagesOneIsPrimaryRestMayNotHaveField,
          "http://www.findyourcanopy.com/image303.jpeg",
        ],
      ];

      for (let x = 0; x < testCases.length; x += 1) {
        const [data, image] = testCases[x];
        getPrimaryPropertyImage.mockImplementation(() => image);

        const propertyImage = getComponent({data}).find(PropertyImage);

        expect(getPrimaryPropertyImage).toHaveBeenCalledWith(
          data.propertyImages,
        );
        expect(propertyImage).toHaveProp(
          "source",
          expect.objectContaining({
            uri: image,
          }),
        );
      }
    });

    it("displays a default image if there're no images", () => {
      const data = getMockProperties().hasNoImages;
      getPrimaryPropertyImage.mockImplementation(() => null);
      expect(
        getComponent({data}).find(StyledDefaultPropertyThumbnail),
      ).toExist();
    });
  });

  describe("Property Settlement", () => {
    it("displays town when there's no city", () => {
      const data = getMockProperties().hasTownNoCity;
      const secondAddressLine = getComponent({data})
        .find(PropertyAddressLine)
        .at(1);
      expect(secondAddressLine).toHaveProp("children", data.address.town);
    });

    it("displays city when there's no town", () => {
      const data = getMockProperties().hasCityNoTown;
      const secondAddressLine = getComponent({data})
        .find(PropertyAddressLine)
        .at(1);
      expect(secondAddressLine).toHaveProp("children", data.address.city);
    });

    it("displays city when there's both city and town", () => {
      const data = getMockProperties().hasCityAndTown;
      const secondAddressLine = getComponent({data})
        .find(PropertyAddressLine)
        .at(1);
      expect(secondAddressLine).toHaveProp("children", data.address.city);
    });
  });

  describe("Property Rental Period", () => {
    it("displays weekly rental period abbreviation", () => {
      const data = getMockProperties().hasWeeklyRent;
      const translations = getMockTranslations({
        rental_period_abbreviations: {weekly: "weekly"},
      });
      const rent = getComponent({data, translations}).find(PropertyRent);
      expect(rent.prop("children")).toEqual(
        expect.stringContaining(
          translations.rental_period_abbreviations.weekly,
        ),
      );
    });

    it("displays fortnightly rental period abbreviation", () => {
      const data = getMockProperties().hasFortnightlyRent;
      const translations = getMockTranslations({
        rental_period_abbreviations: {fortnightly: "fortnightly"},
      });
      const rent = getComponent({data, translations}).find(PropertyRent);
      expect(
        rent
          .prop("children")
          .endsWith(translations.rental_period_abbreviations.fortnightly),
      ).toBeTruthy();
    });

    it("displays four weekly rental period abbreviation", () => {
      const data = getMockProperties().hasFourWeeklyRent;
      const translations = getMockTranslations({
        rental_period_abbreviations: {four_weekly: "four_weekly"},
      });
      const rent = getComponent({data, translations}).find(PropertyRent);
      expect(
        rent
          .prop("children")
          .endsWith(translations.rental_period_abbreviations.four_weekly),
      ).toBeTruthy();
    });

    it("displays monthly rental period abbreviation", () => {
      const data = getMockProperties().hasMonthlyRent;
      const translations = getMockTranslations({
        rental_period_abbreviations: {monthly: "monthly"},
      });
      const rent = getComponent({data, translations}).find(PropertyRent);
      expect(
        rent
          .prop("children")
          .endsWith(translations.rental_period_abbreviations.monthly),
      ).toBeTruthy();
    });

    it("displays yearly rental period abbreviation", () => {
      const data = getMockProperties().hasYearlyRent;
      const translations = getMockTranslations({yearly: "yearly"});
      const rent = getComponent({data, translations}).find(PropertyRent);
      expect(
        rent
          .prop("children")
          .endsWith(translations.rental_period_abbreviations.yearly),
      ).toBeTruthy();
    });
  });

  describe("Item click", () => {
    it("fires click handler on item click", () => {
      const properties = getMockPropertiesAsArray();
      for (let x = 0; x < properties.length; x += 1) {
        const data = properties[x];
        const itemClick = jest.fn();
        const wrapper = getComponent({data, itemClick});
        wrapper.simulate("press");
        expect(itemClick).toHaveBeenCalledWith(data.id);
      }
    });

    it("doesn't crash if no itemClick provided", () => {
      const properties = getMockPropertiesAsArray();
      for (let x = 0; x < properties.length; x += 1) {
        const data = properties[x];
        const wrapper = getComponent({data, itemClick: undefined});
        expect(() => wrapper.simulate("press")).not.toThrow();
      }
    });

    it("wraps component in TouchableOpacity if itemClick provided", () => {
      expect(getComponent().type()).toBe(TouchableOpacity);
    });

    it("doesn't wrap component in TouchableOpacity if no itemClick provided", () => {
      expect(getComponent({itemClick: undefined}).type()).toBe(
        PropertyItemWrapper,
      );
    });
  });

  const getComponent = props => {
    const parsedProps = extend(
      {
        data: getMockPropertiesAsArray()[0],
        translations: getMockTranslations(),
        itemClick: () => {},
      },
      props,
    );

    return shallow(<PropertyItem {...parsedProps} />);
  };
});
