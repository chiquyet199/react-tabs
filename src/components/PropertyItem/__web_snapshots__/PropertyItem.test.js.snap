// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Property item tests PropertyItem snapshot test on - web 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onPress={[Function]}
>
  <Connect(WithTheme(Styled(View)))>
    <Styled(Image)
      source={
        Object {
          "uri": "http://www.findyourcanopy.com/img124.jpeg",
        }
      }
    />
    <Styled(View)>
      <View>
        <Connect(WithTheme(Styled(Text)))>
          29 Acacia Road
        </Connect(WithTheme(Styled(Text)))>
      </View>
      <View>
        <Connect(WithTheme(Styled(Text)))>
          Beanotown
        </Connect(WithTheme(Styled(Text)))>
      </View>
      <View>
        <Connect(WithTheme(Styled(Text)))>
          BE1 1BE
        </Connect(WithTheme(Styled(Text)))>
      </View>
    </Styled(View)>
    <Connect(WithTheme(Styled(Text)))>
      Â£1234.99w
    </Connect(WithTheme(Styled(Text)))>
  </Connect(WithTheme(Styled(View)))>
</TouchableOpacity>
`;
